x/twap/api_test.go

```diff
@@ -254,6 +288,71 @@ func (s *TestSuite) TestGetArithmeticTwap() {
 	}
 }
 
+func (s *TestSuite) TestGetArithmeticTwap_ThreeAsset() {
+	tests := map[string]struct {
+		recordsToSet []types.TwapRecord
+		ctxTime      time.Time
+		input        []getTwapInput
+		expTwap      []sdk.Dec
+		expectError  error
+	}{
+		"(2 pairs of 3 records) start exact, end after second record": {
+			recordsToSet: []types.TwapRecord{tapRecordAB, tapRecordAC, tapRecordBC,
+				tPlus10sp5TapRecordAB, tPlus10sp5TapRecordAC, tPlus10sp5TapRecordBC},
+			ctxTime: tPlusOneMin,
+			input:   makeSimpleTapTwapInput(baseTime, baseTime.Add(20*time.Second), baseQuoteAB, baseQuoteCA, baseQuoteBC),
+			// A 10 for 10s, 5 for 10s = 150/20 = 7.5
+			// C 20 for 10s, 10 for 10s = 300/20 = 15
+			// B .1 for 10s, .2 for 10s = 3/20 = 0.15
+			expTwap: []sdk.Dec{sdk.NewDecWithPrec(75, 1), sdk.NewDec(15), sdk.NewDecWithPrec(15, 2)},
+		},
+		"(3 pairs of 3 record) start at second record, end after third record": {
+			recordsToSet: []types.TwapRecord{tapRecordAB, tapRecordAC, tapRecordBC,
+				tPlus10sp5TapRecordAB, tPlus10sp5TapRecordAC, tPlus10sp5TapRecordBC,
+				tPlus20sp2TapRecordAB, tPlus20sp2TapRecordAC, tPlus20sp2TapRecordBC},
+			ctxTime: tPlusOneMin,
+			input:   makeSimpleTapTwapInput(baseTime.Add(10*time.Second), baseTime.Add(30*time.Second), baseQuoteAB, baseQuoteCA, baseQuoteBC),
+			// A 5 for 10s, 2 for 10s = 70/20 = 3.5
+			// C 10 for 10s, 4 for 10s = 140/20 = 7
+			// B .2 for 10s, .5 for 10s = 7/20 = 0.35
+			expTwap: []sdk.Dec{sdk.NewDecWithPrec(35, 1), sdk.NewDec(7), sdk.NewDecWithPrec(35, 2)},
+		},
+		// interpolate in time closer to second record
+		"(3 record) interpolate: get twap closer to second record": {
+			recordsToSet: []types.TwapRecord{tapRecordAB, tapRecordAC, tapRecordBC,
+				tPlus10sp5TapRecordAB, tPlus10sp5TapRecordAC, tPlus10sp5TapRecordBC,
+				tPlus20sp2TapRecordAB, tPlus20sp2TapRecordAC, tPlus20sp2TapRecordBC},
+			ctxTime: tPlusOneMin,
+			input:   makeSimpleTapTwapInput(baseTime.Add(15*time.Second), baseTime.Add(30*time.Second), baseQuoteAB, baseQuoteCA, baseQuoteBC),
+			// A 5 for 5s, 2 for 10s = 45/15 = 3
+			// C 10 for 5s, 4 for 10s = 140/15 = 6
+			// B .2 for 5s, .5 for 10s = 7/15 = .4
+			expTwap: []sdk.Dec{sdk.NewDec(3), sdk.NewDec(6), sdk.NewDecWithPrec(4, 1)}, // 5 for 5s, 2 for 10s = 45 / 15 = 3
```

Comment:
nit: I think that a copy of this is already present above
```suggestion
			expTwap: []sdk.Dec{sdk.NewDec(3), sdk.NewDec(6), sdk.NewDecWithPrec(4, 1)},
```